src/System.Device.Gpio.Tests/GpioControllerTestBase.cs

```diff
@@ -130,30 +131,203 @@ public void OpenPinDefaultsModeToInput()
         [Fact]
         public void AddCallbackTest()
         {
-            while (!Debugger.IsAttached)
-                Thread.Sleep(1000);
-            Debugger.Break();
-
-            ManualResetEvent mre = new ManualResetEvent(false);
             bool wasCalled = false;
             using (GpioController controller = new GpioController(GetTestNumberingScheme(), GetTestDriver()))
             {
                 controller.OpenPin(InputPin, PinMode.Input);
                 controller.OpenPin(OutputPin, PinMode.Output);
                 controller.RegisterCallbackForPinValueChangedEvent(InputPin, PinEventTypes.Rising, callback);
                 controller.Write(OutputPin, PinValue.High);
-                mre.WaitOne(TimeSpan.FromSeconds(5));
+                Thread.Sleep(10);
```

Comment:
how did we pick 10?  I'm worried about having a flaky/unreliable test here if the value is too small...

Reply:
Not really sure how to choose this value, by my understanding it is really hardware dependent.
So I first used 100 millisec, just thought it will be enough, Krzysztof suggested to lower it, so changed to 10, worked for the Pi am using ...

Reply:
I think for direct GPIO writes 1us is what we theoretically should aim for but since we are not running on real-time OS 10ms sounds reasonable. For events I don't have strong opinion but I think at the current state I wouldn't expect anything less than 12ms (approximate Thread.Sleep resolution) to work reliably

Reply:
Now using one variable for all sleep duration, will see how it works on repo test server and update accordingly